var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -67,68 +70,93 @@ def setup_dependent_package(self, module, dep_spec):
-                    # and using LD_LIBRARY_PATH throughout Spack.
-                    f.write('[mkl]\n')
-                    f.write('mkl_libs=%s\n'     % 'mkl_rt')
-                elif '^atlas' in spec:
-                    f.write('[atlas]\n')
-                    f.write('atlas_libs=%s\n'   % names)
+            lapackblas_includes.add(spec['blas'].prefix.include)
+            lapackblas_libraries += spec['blas'].blas_libs
+
+        # Each BLAS/LAPACK requires a custom site.cfg
+
+        # Generic build notes (very out-of-date):
+        # http://www.scipy.org/scipylib/building/linux.html
+
+        # Example site.cfg:
+        # https://github.com/numpy/numpy/blob/master/site.cfg.example
+        with open('site.cfg', 'w') as f:
+            libraries = ','.join(lapackblas_libraries.names)
+
+            if '^openblas' in spec:
+                f.write('[openblas]\n')
+                f.write('libraries = {0}\n'.format(libraries))
+
+            elif '^mkl' in spec:
+                # Intel MKL build notes:
+                # https://software.intel.com/en-us/articles/numpyscipy-with-intel-mkl
+
+                # library_dirs are in compilers_and_libraries_2017/linux/mkl/lib/intel64  # noqa
+                # include_dirs are in compilers_and_libraries_2017/linux/mkl/include      # noqa
+                lapackblas_includes = set()
+                for libdir in lapackblas_libraries.directories:
+                    incdir = os.path.dirname(os.path.dirname(libdir))
+                    incdir = join_path(incdir, 'include')
+                    lapackblas_includes.add(incdir)
+
+                f.write('[mkl]\n')
+                f.write('mkl_libs = {0}\n'.format('mkl_rt'))
+                f.write('lapack_libs = \n')
+
+            elif '^atlas' in spec:
+                f.write('[atlas]\n')
+                f.write('atlas_libs = {0}\n'.format(libraries))
+
+            else:
+                # The section title for the defaults changed in @1.10
+                if spec.satisfies('@1.10:'):
+                    f.write('[ALL]\n')
                 else:
-                    # The section title for the defaults changed in @1.10, see
-                    # https://github.com/numpy/numpy/blob/master/site.cfg.example
-                    if spec.satisfies('@:1.9.2'):
-                        f.write('[DEFAULT]\n')
-                    else:
-                        f.write('[ALL]\n')
-                    f.write('libraries=%s\n'    % names)
-
-                f.write('library_dirs=%s\n' % dirs)
-                if not ((platform.system() == "Darwin") and
-                        (platform.mac_ver()[0] == '10.12')):
-                    f.write('rpath=%s\n' % dirs)
+                    f.write('[DEFAULT]\n')
+                f.write('libraries = {0}\n'.format(libraries))
+
+            include_dirs = os.pathsep.join(lapackblas_includes)
+            library_dirs = os.pathsep.join(lapackblas_libraries.directories)
```

Comment:
`os.pathsep` _is_ `:`, on most Unix platforms:
https://docs.python.org/2/library/os.html#os.pathsep

Apparently some use `,` though?? I decided to make this change because that's what numpy specifically recommended:
https://github.com/numpy/numpy/blob/master/site.cfg.example#L27